### PR TITLE
elf: fix parsing of symtab from viv data

### LINKED ISSUE #1704: elf: ValueError: symbol name not found
`libyvr.so` from https://www.virustotal.com/gui/file/4261cc05a8c4ecaf1605ef931397a4d97cc12fe38738a4f6016c3695aa2c571f/detection

```
> python -m capa.main --os linux  -r rules libyvr.so
Traceback (most recent call last):
  File "C:\Users\spring\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\spring\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Exclusions\capa\capa\main.py", line 1345, in <module>
    sys.exit(main())
  File "C:\Exclusions\capa\capa\main.py", line 1275, in main
    capabilities, counts = find_capabilities(rules, extractor, disable_progress=args.quiet)
  File "C:\Exclusions\capa\capa\main.py", line 283, in find_capabilities
    function_matches, bb_matches, insn_matches, feature_count = find_code_capabilities(
  File "C:\Exclusions\capa\capa\main.py", line 210, in find_code_capabilities
    features, bmatches, imatches = find_basic_block_capabilities(ruleset, extractor, fh, bb)
  File "C:\Exclusions\capa\capa\main.py", line 166, in find_basic_block_capabilities
    ifeatures, imatches = find_instruction_capabilities(ruleset, extractor, f, bb, insn)
  File "C:\Exclusions\capa\capa\main.py", line 133, in find_instruction_capabilities
    for feature, addr in itertools.chain(
  File "C:\Exclusions\capa\capa\features\extractors\viv\extractor.py", line 77, in extract_insn_features
    yield from capa.features.extractors.viv.insn.extract_features(fh, bbh, ih)
  File "C:\Exclusions\capa\capa\features\extractors\viv\insn.py", line 737, in extract_features
    for feature, addr in insn_handler(f, bb, insn):
  File "C:\Exclusions\capa\capa\features\extractors\viv\insn.py", line 125, in extract_insn_api_features
    sym_name = symtab.get_name(symbol)
  File "C:\Exclusions\capa\capa\features\extractors\elf.py", line 702, in get_name
    raise ValueError("symbol name not found")
ValueError: symbol name not found
```

### PR DESCRIPTION
**Pull request by williballenthin**
closes #1704 

also cleans up naming and type hints

### Checklist

- [ ] No CHANGELOG update needed

- [x] No new tests needed

- [x] No documentation update needed

### CODE REVIEW

File: capa/features/extractors/elf.py [hunk 1]
```diff
@@ -710,17 +712,17 @@ def get_symbols(self) -> Iterator[Symbol]:
...
+        endian = "<" if elf.getEndian() == 0 else ">"
+        bitness = elf.bits
 
         SHT_SYMTAB = 0x2
-        for section in ElfBinary.sections:
-            if section.sh_info & SHT_SYMTAB:
-                strtab_section = ElfBinary.sections[section.sh_link]
-                sh_symtab = Shdr.from_viv(section, ElfBinary.readAtOffset(section.sh_offset, section.sh_size))
+        for section in elf.sections:
+            if section.sh_type & SHT_SYMTAB:
```
Comment (williballenthin) [inline capa/features/extractors/elf.py:721]: this is the bug fix: use `sh_type` not `sh_info`.

File: capa/features/extractors/elf.py [hunk 2]
```diff
@@ -710,17 +712,17 @@ def get_symbols(self) -> Iterator[Symbol]:
         yield from self.symbols
 
     @classmethod
-    def from_Elf(cls, ElfBinary) -> Optional["SymTab"]:
-        endian = "<" if ElfBinary.getEndian() == 0 else ">"
-        bitness = ElfBinary.bits
+    def from_viv(cls, elf: Elf.Elf) -> Optional["SymTab"]:
```
Comment (williballenthin) [inline capa/features/extractors/elf.py:715]: updated:
  - name for consistency
  - parameter name for pep8 (lowercase)
  - type hint for mypy

File: capa/features/extractors/elf.py (hunk 1 above)
Comment (williballenthin) [reply capa/features/extractors/elf.py:721]:
```suggestion
            if section.sh_type == SHT_SYMTAB:
```

and we should use exact matching, not bit masking here.

**Review by mr-tz — APPROVED** [review]
nice, thanks!

**Comment by yelhamer** [comment]
The failing tests are relevant to the the symbol extraction. The `connect` symbol (and others) aren't being extracted with this change, so we'd probably want to keep extracting the symbol's table using the sh_info field?

I'll take a further look at this in a while.

**Comment by williballenthin** [comment]
ah, good call that the tests are failing 🤦🏼 good thing we have those!

tbh, i dont understand how the existing logic worked, except perhaps by random chance (random number & 2 > 0 is true 50% of the time). i'll also dig into the failing tests and suggest further fixes.

**Comment by williballenthin** [comment]
with [737fab7](https://github.com/mandiant/capa/pull/1721/commits/737fab7969b415ae0550f0ed361bcf74e8a40577) tests are working for me again locally

**Comment by williballenthin** [comment]
awaiting for tests to pass in CI and then i'll merge.

**Pull request merged**